cli/command/cli.go

```diff
@@ -45,7 +45,7 @@ type DockerCli struct {
 func (cli *DockerCli) HasExperimental() bool {
 	if cli.hasExperimental == nil {
 		if cli.client == nil {
-			cli.Initialize(cliflags.NewClientOptions())
+			return false
 		}
 		enabled := false
 		cli.hasExperimental = &enabled
```

Comment:
Altneratively, we could keep `HasExperimental()` but make it only return the boolean, and not store anything. So we could still remove `hasExperimental`